src/routes/expenses.js

```diff
@@ -0,0 +1,128 @@
+'use strict';
+
+const express = require('express');
+const router = express.Router();
+const { users } = require('./users.js');
+
+let expenses = [];
+
+router.get('/', (req, res) => {
+  const {
+    userId,
+    from,
+    to,
+    categories,
+  } = req.query;
+
+  let filteredExpenses = expenses;
+
+  if (userId) {
+    filteredExpenses = filteredExpenses
+      .filter(expense => expense.userId === +userId);
+  }
+
+  if (from && to) {
+    const fromDate = new Date(from);
+    const toDate = new Date(to);
+
+    filteredExpenses = expenses.filter(expense => {
+      const spentAt = new Date(expense.spentAt);
+
+      return spentAt >= fromDate && spentAt <= toDate;
+    });
+  }
+
+  if (userId && categories) {
+    filteredExpenses = expenses.filter(expense => (
+      categories.includes(expense.category)
+    ));
+  }
+
+  res.send(filteredExpenses);
```

Comment:
Here and in other places, the array is returned to a straight line, although the JSON object should be returned.